### PR TITLE
[#557] de-duplicate acl lists in case of multiple replicas.

### PR DESCRIPTION
**Pull request by d-w-moore**

### CODE REVIEW

**Comment by trel** [comment]
this is 557, right?

File: irods/manager/access_manager.py [hunk 1]
```diff
@@ -116,12 +116,15 @@ def __get_raw(self, target, **kw):
         elif users_out is None: pass
         else:                   raise TypeError
 
-        acls = [ iRODSAccess ( r[access_column.name],
+        # Instanstiate as set before converting to a list, in order to remove duplicate iRODSAccess
```
Comment (trel) [inline irods/manager/access_manager.py:119]: Instanstiate -> Instantiate

File: irods/access.py [hunk 2]
```diff
@@ -63,6 +64,18 @@ def __init__(self, access_name, path, user_name='', user_zone='', user_type=None
...
+               iRODSPath(self.path) == iRODSPath(other.path) and \
+               self.user_name == other.user_name and \
+               self.user_zone == other.user_zone
+
+    def __hash__(self):
+        return hash((self.access_name,
+                     iRODSPath(self.path),
+                     self.user_name,
+                     self.user_zone))
+
```
Comment (trel) [inline irods/access.py:78]: i don't see these... used?

**Review by alanking — COMMENTED** [review]
Seems good to me. Yes, let's fix the typo and put an issue number on it

**Comment by alanking** [comment]
Oh. Also, please add a test for this.

**Comment by d-w-moore** [comment]
> this is 557, right?

Yes - just now amended title and commit message to reflect that.

File: irods/access.py (hunk 2 above)
Comment (d-w-moore) [reply irods/access.py:78]: correct

**Comment by d-w-moore** [comment]
Test added, conversations resolved.

File: irods/test/data_obj_test.py [hunk 3]
```diff
@@ -129,6 +129,19 @@ def create_resc_hierarchy (self, Root, Leaf = None):
             self.sess.resources.remove(Leaf)
             self.sess.resources.remove(Root)
 
+    def test_acls_are_uniquely_listed_for_replicated_data_objects__issue_557(self):
+        with self.create_simple_resc() as newResc1:
+            d = None
+            try:
+                path = self.coll_path + "/" + unique_name (my_function_name(), datetime.now())
```
Comment (trel) [inline irods/test/data_obj_test.py:136]: space after unique_name

**Comment by trel** [comment]
looks good enough to squash.  no # please.

**Comment by d-w-moore** [comment]
Squashed

**Review by alanking — APPROVED** [review]
Great. \# it!

**Comment by alanking** [comment]
@d-w-moore - I think we're just waiting on #s for this one

**Comment by d-w-moore** [comment]
> Great. # it!

pound added.

**Pull request merged**